internal/batches/service/service.go

```diff
@@ -755,7 +763,7 @@ func (svc *Service) FindDirectoriesInRepos(ctx context.Context, fileName string,
 
 var defaultQueryCountRegex = regexp.MustCompile(`\bcount:(\d+|all)\b`)
 
-const hardCodedCount = " count:999999"
+const hardCodedCount = " count:99999"
```

Comment:
This is meant to emulate `count:all` which didn't exist back then. In order not to change the behavior here (probably only for the biggest customers, but still) I think we should not tune down this number. That's just how BC is documented to operate on the search. 
I assume this is mainly a problem on dotcom with its massive scale and that's now already covered by the above change, isn't it?

Reply:
Gotcha gotcha, that makes sense. The concern is just that our biggest customers might *also* hit timeout issues like the user reported in the original issue if their query actually approaches a results count of 999999, and we can't easily detect that and tell the user when it's happening, so it ends up just looking like flakiness instead. You're probably right that dotcom is still going to top the instances in terms of scale, though.

Related to this, though, how did we settle on the initial number of 999k as an emulation of `count:all`? How do we know 999k is "sufficiently large", 9M is too big, and 99999 is actually not large enough? Does search cap out at a certain count?

Reply:
I think we arbitrarily hit the 9 key until it felt ludicrously large to be honest. That was when sourcegraph dotcom was on the order of 10k repos indexed. I've wanted to adopt the count:all flag instead forever but it isn't so long until we don't need this anymore at all because SSBC does use count:all so .. I guess not worth breaking our heads over this for too long 😅

Reply:
Haha okay fair! I'll leave it unchanged for now, then, since we haven't heard complaints about this from enterprise customers yet and it sounds like once we use `count:all` this would also be a non-issue. 🙂 Thanks Erik!